src/net/gateway-pool.ts

```diff
@@ -39,10 +39,7 @@ export class GatewayPool {
     static discover(params: IGatewayDiscoveryParameters): Promise<GatewayPool> {
         return GatewayDiscovery.run(params)
             .then((bestNodes: Gateway[]) => {
-                const pool = new GatewayPool(bestNodes, params)
-
-                return pool.init()
-                    .then(() => pool)
+                return new GatewayPool(bestNodes, params)
```

Comment:
@marcvelmer  Why do you avoid the init inside the discover function?

Reply:
Because the init is calling the init function of the active gateway, which ends up calling the init function of the dvote gateway. There the function isUp() is called, which is already being called before, in the run function of the discovery.
I found this by debugging the API calls made by the `getInfo()` function of the dvote gateway and thought that maybe we could save one API call by avoiding calling the init function of the pool at this point.

Reply:
@marcvelmer  Nice catch :-)  I agree we should remove the duplicate isUp call, but I think though what we should modify is what is done during the init function

Reply:
@marcvelmer maybe we could use the isReady function:
https://github.com/vocdoni/dvote-js/blob/4ff00fac504233fbee87e78641ba4f923d6677e8/src/net/gateway.ts#L135
And if isReady returns false then do a gateway refresh in the catch

Reply:
I changed the behaviour of what is done inside the `init()` function in order to not duplicate a `isUp()` call, but I still need some review.

Reply:
This will be reviewed again when refactoring the pool.